### PR TITLE
add binding of config variables

### PR DESCRIPTION
**Pull request by rfomin**
A system like in Chocolate Doom for config variables, so we can potentially reduce the number of global variables and achieve better "encapsulation". I also think it's more convenient to bind config variables in a module rather than in m_config.c.

Should I continue?

### CODE REVIEW

**Comment by fabiangreffrath** [comment]
I like this approach, but how dows it help to reduce the number of global variables?

**Comment by rfomin** [comment]
> I like this approach, but how dows it help to reduce the number of global variables?

Theoretically, we could make all sound module variables `static` and remove them from the `.h` files. The same applies to the video and input variables.

**Comment by fabiangreffrath** [comment]
Ah, right!

**Comment by rfomin** [comment]
Maybe I overdid it. @fabiangreffrath What do you think?

**Comment by fabiangreffrath** [comment]
What do you mean? Well, the patch is super invasive, but we knew that before, but the implementation is as clean as it can get.

File: src/i_video.c [hunk 1]
```diff
@@ -1779,6 +1786,43 @@ void I_InitGraphics(void)
...
+    BIND_BOOL(fullscreen, true, "1 to enable fullscreen");
+    BIND_BOOL(exclusive_fullscreen, false, "1 to enable exclusive fullscreen");
+    BIND_BOOL_GEN(use_vsync, true,
+        "1 to enable wait for vsync to avoid display tearing");
+    BIND2_BOOL_GEN(uncapped, true, "1 to enable uncapped rendering frame rate");
+    BIND_NUM_GEN(fpslimit, 0, 0, 500,
+        "Framerate limit in frames per second (< 35 = disable)");
+    BIND2_NUM_GEN(widescreen, RATIO_AUTO, 0, NUM_RATIOS - 1,
+        "Widescreen (0 = Off, 1 = Auto, 2 = 16:10, 3 = 16:9, 4 = 21:9)");
+    M_BindNum("fov", &custom_fov, NULL, FOV_DEFAULT, FOV_MIN, FOV_MAX,
```
Comment (fabiangreffrath) [inline src/i_video.c:1804]: The changes in `M_BindNum()`, `BIND_NUM_GEN()`, `BIND2_NUM()`, etc. are confusing.

Comment (rfomin) [reply src/i_video.c:1804]: So should I get rid of these macros? I think we can keep the most useful ones `BIND_NUM`, `BIND_BOOL`, `BIND_NUM_GEN` and `BIND_BOOL_GEN`. (`GEN` refers to the "General" options menu).

Comment (fabiangreffrath) [reply src/i_video.c:1804]: Yes, probably. It doesn't help to have that many macros with similar names but rather specific uses. Also, could you rename `GEN` to `GENERAL`?

Comment (rfomin) [reply src/i_video.c]: Done in https://github.com/fabiangreffrath/woof/pull/1677/commits/c00fcc35c51b941a95111df3372ef0334cc2f58c

**Review by fabiangreffrath — APPROVED** [review]
Thank you! 

Should we now let @MrAlaux review the cvar descriptions?

**Comment by rfomin** [comment]
> Should we now let @MrAlaux review the cvar descriptions?

Yes, sure.

**Pull request merged**